bfsfs/bfsfs.go

```diff
@@ -32,8 +32,11 @@ import (
 
 func init() {
 	bfs.Register("file", func(_ context.Context, u *url.URL) (bfs.Bucket, error) {
-		q := u.Query()
-		return New(u.Path, q.Get("tmpdir"))
```

Comment:
that's a breaking change, no?

Reply:
I generally don't disagree, we do use `prefix=...` param for s3 and gs but not sure how this will affect existing users

Reply:
Well, yes, it's _kind of_ breaking. But it looks like I'm the only one using this :)

But seriously - even if it's used in other places, it's (probably) used with some custom code to make it work (it contradicts bfsgs/bfss3 usage pattern), which is totally wrong.

It shouldn't have any custom code to make it work.

Ideally, we'd better have global `Open(...)` and other `Bucket.*` methods instead of messing with `Resolve()`, but it's a bit too much, I think.

Reply:
I share your concern - it looks weird that path is split (well, URL - host/path), but it really helps a lot.

Reply:
@mxmCherry please let me think about this for a day. if we change this (requires new major version), I also propose that we change the way the other resolves work,

Reply:
If it helps - looks like most "popular" usage pattern is to:

- use `gs://...` or `s3://...` for prod
- use `mem://` for testing (don't remember, but I guess each test suite requires Register-ing the scheme resolver which returns pre-made in-mem bucket)

I doubt anyone would ever use `mem://` for non-testing purposes.

I'd suggest the API to look like:

```
import (
  "github.com/bsm/bfs"

  // needed "drivers" - we can even make their contents all-private and get rid of "bfsXX" prefixes as these won't ever be referenced in caller code:
  _ "github.com/bsm/bfs/bfsgs" // gs://
  _ "github.com/bsm/bfs/bfss3" // s3://
  _ "github.com/bsm/bfs/bfsfs" // file://
  _ "github.com/bsm/bfs/bfsmem" // mem:// - could be part of bfs itself, but I'd like to make it explicit
)

// all API methods/funcs are just package-level:
f, err := bfs.Create(ctx, "mem://path/to/file.bin")
f, err := bfs.Open(ctx, "mem://path/to/file.bin")
it, err := bfs.Glob(ctx, "mem://path/to/dir/*")
// and so on
```

Make it less flexible, but cover all the use-cases with minimum caller code.

Reply:
With such API I'd even call this not "bfs" (bucket-file-system) but something like "abstractfs" or so (longer, but that's what it is - FS abstraction).

Reply:
Oh, another feature request would be - let's have `http(s)://` scheme support at least for Open? Sometimes it's useful to read HTTP files using same abstraction (again, mostly for testing). With such adapter other methods could return error like "not supported".